### PR TITLE
Don't crash when netlink interface is not present (for vfio devices)

### PR DESCRIPTION
**Pull request by booxter**

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -140,18 +140,19 @@ func GetSharedPF(ifName string) (string, error) {
...
 // GetVFLinkNames returns VF's network interface name given it's PF name as string and VF id as int
 func GetVFLinkNames(pfName string, vfID int) ([]string, error) {
-	var names []string
+	var names = make([]string, 0)
+
 	vfDir := filepath.Join(NetDirectory, pfName, "device", fmt.Sprintf("virtfn%d", vfID), "net")
 	if _, err := os.Lstat(vfDir); err != nil {
-		return nil, err
+		// if net directory is missing the device is probably unregistered from its network driver
+		return names, nil
```
Comment (ahalim-intel) [inline pkg/utils/utils.go]: With netdevice we need be able to catch this error.  As @zshi-redhat indicated, this is going to be a temp fix for now, and need to be reverted back with newer dpdk support.

**Comment by ahalim-intel** [comment]
could you fix the unit test so that all check passes?

**Comment by ahalim-intel** [comment]
The commit [79b2a6d](/intel/sriov-cni/pull/64/commits/79b2a6de391d84e280fbfa33b720b62b206fffea) has image tag release-v1 and should not be used for image built from master.

**Comment by booxter** [comment]
@ahalim-intel ok about the commit, but is building an image in master not broken though? That's actually one thing that surprises me in this repo - that it seems like development is happening in a release branch and not in master, and master doesn't get maintained (?). For example, make image is broken.

File: pkg/utils/utils.go (hunk 1 above)
Comment (booxter) [reply pkg/utils/utils.go]: I actually need to check which driver the VF is attached to, and skip for vfio / uio only.

**Comment by ahalim-intel** [comment]
> @ahalim-intel ok about the commit, but is building an image in master not broken though?
For exact same reason above we could not apply the same patch for master because it had release-v1 tag. In fact ,release-v1 fixes cannot be applied directly to master without modification.  I personally, do not build sriov cni image, I rather work with built binary, its quicker to test out stuff. This wasn't a priority. If someone have a patch we will happily apply. Thanks

**Comment by booxter** [comment]
@ahalim-intel what I am saying is that common practice is having all fixes flow into master and then backporting them to release branches, but in this repo it seems like these two branches are independent and don't share the code, so one can have a fix while another may miss it. Do you have plans to merge back in the future?

**Comment by ahalim-intel** [comment]
There's no plan of merging these two branches. All new developments/features will go in the master as usual. New features won't be added to the release-v1. We might have new release branch with new features. We will fix any known bugs to all branches and should try to apply bug fixes on all branch if possible, otherwise need to apply branch specific fix.  These two branches have been diverted already.

**Comment by booxter** [comment]
I am fixing the master branch image builds here: https://github.com/intel/sriov-cni/pull/65

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -15,6 +15,8 @@ var (
 	NetDirectory = "/sys/class/net"
 	// SysBusPci is sysfs pci device directory
 	SysBusPci = "/sys/bus/pci/devices"
+	// UserspaceDrivers is a list of driver names that don't have netlink representation for their devices
+	UserspaceDrivers = []string{"vfio-pci", "uio_pci_generic"}
```
Comment (booxter) [inline pkg/utils/utils.go]: I *think* that's the right name for UIO driver but I am not sure. Perhaps someone with uio experience can clarify.

Comment (ahalim-intel) [reply pkg/utils/utils.go]: It is right one. We also need to include "igb_uio" there.

**Comment by zshi-redhat** [comment]
/lgtm once adding ``igb_uio`` in ``userspaceDrivers``
tested ``vfio-pci`` in my local environment, the dpdk Pod can be created successfully with this PR applied.

**Pull request merged**